parsing/location.ml

```diff
@@ -223,9 +223,29 @@ let rec highlight_locations ppf locs =
 
 open Format
 
+let rewrite_absolute_path =
+  let init = ref false in
+  let map_cache = ref None in
+  fun path ->
+    if not !init then begin
+      init := true;
+      match Sys.getenv "BUILD_PATH_PREFIX_MAP" with
+      | exception Not_found -> ()
+      | encoded_map ->
+        match Build_path_prefix_map.decode_map encoded_map with
+          | Error err -> failwith err (* TODO improve *)
```

Comment:
IIUC, this `failwith` means if I launch the compiler with an ill-formed value for `BUILD_PATH_PREFIX_MAP` I might get a compiler fatal error with a message like:
```
Fatal error: Failure "invalid character '=' in key or value"
```
This is not nice. The user-visible error message must at least mention the environment variable name.